### ISSUE TITLE
`brew prof` fails because of Ruby version

### ISSUE THREAD

**Issue by dnicolson** [Homebrew/brew#6437]
- [x] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [x] ran a `brew` command and reproduced the problem with multiple formulae? If it's a problem with a single, official formula (not cask) please file this issue at Homebrew/homebrew-core: https://github.com/Homebrew/homebrew-core/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [ ] ran `brew config` and `brew doctor` and included their output with your issue?

# What you were trying to do (and why)
Run the following command:
`brew prof uses wget`

# What happened (include command output)
```
==> Installing 'ruby-prof' gem
ERROR:  Error installing ruby-prof:
	ruby-prof requires Ruby version >= 2.4.0.
Error: failed to install the 'ruby-prof' gem.
```

<details>
  <summary>brew doctor</summary>
  <pre>
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and which additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew-provided
script of the same name. We found the following "config" scripts:
  /Users/dnicolson/.asdf/shims/python2-config
  /Users/dnicolson/.asdf/shims/python3.6m-config
  /Users/dnicolson/.asdf/shims/python2.7-config
  /Users/dnicolson/.asdf/shims/python-config
  /Users/dnicolson/.asdf/shims/python3-config
  /Users/dnicolson/.asdf/shims/python3.6-config
  </pre>
</details>

<details>
  <summary>brew config</summary>
  <pre>
 HOMEBREW_VERSION: 2.1.11-51-g4a3442d
ORIGIN: https://github.com/Homebrew/brew
HEAD: 4a3442d1a4b3a6f99b74d3740853447d94261eac
Last commit: 3 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: bf7c445cf3a34fc0efe50cdf29c95ea48d265c50
Core tap last commit: 11 minutes ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_DEV_CMD_RUN: 1
CPU: quad-core 64-bit kabylake
Homebrew Ruby: 2.3.7 => /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/bin/ruby
Clang: 10.0 build 1001
Git: 2.22.0 => /usr/local/bin/git
Curl: 7.54.0 => /usr/bin/curl
Java: 1.8.0_222, 1.8.0_202
macOS: 10.14.6-x86_64
CLT: 10.3.0.0.1.1562985497
Xcode: 10.3
CLT headers: 10.3.0.0.1.1562985497
XQuartz: 2.7.11 => /opt/X11
  </pre>
</details>

# What you expected to happen
The command to run without error.

# Step-by-step reproduction instructions (by running `brew` commands)
1. `brew prof uses wget`

**Comment by timorunge** [comment]
Basically the problem is, that `ruby-prof 1.0.0` dropped support for older Ruby versions (see https://ruby-prof.github.io/#version-1.0).

This patch will sort out the issue with an older Ruby version for you (on MacOS):

```
diff --git a/Library/Homebrew/dev-cmd/prof.rb b/Library/Homebrew/dev-cmd/prof.rb
index 4666fa0d0..aba6de474 100644
--- a/Library/Homebrew/dev-cmd/prof.rb
+++ b/Library/Homebrew/dev-cmd/prof.rb
@@ -18,7 +18,7 @@ module Homebrew
   def prof
     prof_args.parse
 
-    Homebrew.install_gem_setup_path! "ruby-prof"
+    Homebrew.install_gem_setup_path! "ruby-prof", version: "0.18.0"
     FileUtils.mkdir_p "prof"
     brew_rb = (HOMEBREW_LIBRARY_PATH/"brew.rb").resolved_path
     safe_system "ruby-prof", "--printer=multi", "--file=prof", brew_rb, "--", *ARGV
```

Alternatively you can also use brew to install a newer version of Ruby (`brew install ruby`) and try to run `brew prof uses wget` again (I've not tested this solution).

**Comment by dnicolson** [comment]
There seems to be shell code for determining which version Ruby to use, but it always defaults to the included 2.3.7 vendor version. Using brew, rbenv or asdf don't seem to have any effect.
